### PR TITLE
As/analytics

### PR DESCRIPTION
**Pull request by Apsaliya**

### CODE REVIEW

File: app/src/main/java/wordtextcounter/details/main/util/RateUsHelper.kt [hunk 1]
```diff
@@ -35,27 +38,32 @@ object RateUsHelper {
...
       }
 
       dialog.btnNever.setOnClickListener {
+        logAnalytics(Click("rateus_never"))
         onNoSelected(activity)
         dialog.dismiss()
       }
 
       dialog.setOnKeyListener { _, keyCode, _ ->
+        logAnalytics(Click("rateus_backpress"))
```
Comment (Hirak20) [inline app/src/main/java/wordtextcounter/details/main/util/RateUsHelper.kt]: This sould go in if block

File: app/src/main/java/wordtextcounter/details/main/feature/settings/SettingsFragment.kt [hunk 2]
```diff
@@ -3,18 +3,34 @@ package wordtextcounter.details.main.feature.settings
...
 
 class SettingsFragment : PreferenceFragmentCompat() {
 
   override fun onCreatePreferencesFix(savedInstanceState: Bundle?, rootKey: String?) {
     addPreferencesFromResource(R.xml.settings)
 
+    val clipboardPref = preferenceManager.findPreference(
+        Constants.PREF_CLIPBOARD)
+    clipboardPref.setOnPreferenceChangeListener { _, _ ->
+      logAnalytics(Event("settings_clipboard_change"))
```
Comment (Hirak20) [inline app/src/main/java/wordtextcounter/details/main/feature/settings/SettingsFragment.kt]: How about use pref.key in place of extra strings for all prefs. Will create ext function as we add more settings

File: app/src/main/java/wordtextcounter/details/main/feature/notes/DraftsAdapter.kt [hunk 3]
```diff
@@ -87,6 +90,10 @@ class DraftsAdapter :
       delete.setOnClickListener {
         clickRelay.accept(DraftActions.DraftDelete(itemView.tag as Draft))
       }
+
+      itemView.setOnClickListener {
```
Comment (Hirak20) [inline app/src/main/java/wordtextcounter/details/main/feature/notes/DraftsAdapter.kt]: Do we need this?

File: app/src/main/java/wordtextcounter/details/main/analytics/AnalyticsConsent.kt [hunk 4]
```diff
@@ -28,6 +30,7 @@ object AnalyticsConsent {
     val btSave = dialog.findViewById<Button>(R.id.btSave)
 
     btSave.setOnClickListener {
+      logAnalytics(Click("consent_save"))
```
Comment (Hirak20) [inline app/src/main/java/wordtextcounter/details/main/analytics/AnalyticsConsent.kt]: Not sure if GDPR allows even this logging

File: app/src/main/java/wordtextcounter/details/main/util/RateUsHelper.kt (hunk 1 above)
Comment (Apsaliya) [reply app/src/main/java/wordtextcounter/details/main/util/RateUsHelper.kt]: yep, will update

File: app/src/main/java/wordtextcounter/details/main/feature/settings/SettingsFragment.kt (hunk 2 above)
Comment (Apsaliya) [reply app/src/main/java/wordtextcounter/details/main/feature/settings/SettingsFragment.kt]: not sure I follow.

File: app/src/main/java/wordtextcounter/details/main/analytics/AnalyticsConsent.kt (hunk 4 above)
Comment (Apsaliya) [reply app/src/main/java/wordtextcounter/details/main/analytics/AnalyticsConsent.kt]: That's safe. If consent wasn't given then Logger won't do anything anyway.

File: app/src/main/java/wordtextcounter/details/main/feature/notes/DraftsAdapter.kt (hunk 3 above)
Comment (Apsaliya) [reply app/src/main/java/wordtextcounter/details/main/feature/notes/DraftsAdapter.kt]: yep, why not?

File: app/src/main/java/wordtextcounter/details/main/analytics/AnalyticsConsent.kt (hunk 4 above)
Comment (Apsaliya) [reply app/src/main/java/wordtextcounter/details/main/analytics/AnalyticsConsent.kt]: Now that I think, this feels redundant. This should be called at the end where we dismiss the dialog. wii update.

**Pull request merged**